### ISSUE TITLE
Restrict storage changes in decision lambda

### ISSUE THREAD

**Issue by Martoon-00** [tezos-commons/baseDAO#55]
# Clarification and motivation

Problem: currently `decisionLambda` can modify the storage arbitrarily, and this potentially can violate the invariants of the contract.

The best way to resolve this is to provide a limited `StorageC` constraint into the lambda (fortunately, it already operates on a polymorphic `store`, not the concrete `Storage`). 

For this, we need Morley to support `StoreHasField` and `StoreHasSubmap` in read-only modes.
Though I think this makes sense doing only if the existing instances will work as-is with the new interface, otherwise the resulting complexity does not worth it.

**Note:** whatever you decide to do, make sure your changes do not conflict with [this MR working on `StoreClass.hs`](https://gitlab.com/morley-framework/morley/-/merge_requests/640) in Morley.

# Acceptance criteria

* Morley is updated to support read-only mode in `StoreHasField` and `StoreHasSubmap`.
* `decisionLambda` provides a reduced `StorageC` where we provide access only to ledger, operators and contract extra.
* Methods like `transfer` and related can be invoked from within the decision lambda.

**Comment by pasqu4le** [comment]
Note: the details of the issue above are specific to the Lorentz version, which will be soon dismissed in #169 

I'm keeping this open because the Ligo version is already more restrictive, but that may need to be reconsidered.

**Comment by pasqu4le** [comment]
Update: we should probably avoid discussing this until after #263 

Currently the contract storage is divided in two major blocks, mostly due to supporting lambdas, which won't be required anymore after that.

If that's indeed the case I suspect that we can move `config` under the `storage` and make it so that what's in that (and only that) can be modified by proposals.

**Comment by pasqu4le** [comment]
As expected, after #263 the section of the storage that can be affected by proposals is much more restricted, see [`decision_callback_output`](`https://github.com/tezos-commons/baseDAO/blob/d095368fc9c9fabd1f9a9bb5903efa50d36798a2/src/types.mligo#L253).

Given that this was made to be as small as possible, I'm closing this issue because no longer relevant.
